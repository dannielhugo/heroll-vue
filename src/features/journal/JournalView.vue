<template>
  <div class="journal-view__search">
    <HSearchInput @selected="onAdd" />
  </div>
  <GameJournal :loading="loading" :games="gamesInPage" />
  <HPagination
    class="journal-view__pagination"
    :total="total"
    :loading="loading"
    :page-size="perPage"
    :current-page="page"
    @page-changed="onPageChanged"
  />

  <RouterView />
</template>

<script setup lang="ts">
import GameJournal from '@/components/journal/GameJournal.vue';
import HSearchInput from '@/components/search/HSearchInput.vue';
import HPagination from '@/components/utility/HPagination.vue';
import useGameAdd from '@/composables/journal/use-game-add';
import useGameList from '@/composables/journal/use-game-list';
import useGameListPagination from '@/composables/journal/use-game-list-pagination';

const { loading, games } = useGameList();
const { onAdd } = useGameAdd();
const { gamesInPage, page, total, perPage } = useGameListPagination(games);

const onPageChanged = (newPage: number) => {
  page.value = newPage;
};
</script>

<style scoped lang="scss">
.journal-view {
  &__pagination {
    justify-content: center;
    margin-top: 1rem;
  }

  &__search {
    margin-bottom: 2rem;
    align-self: left;
    width: 33%;

    @media (max-width: 768px) {
      width: 100%;
    }
  }
}
</style>
