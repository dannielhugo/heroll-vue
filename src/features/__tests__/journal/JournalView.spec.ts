import { ref } from 'vue';
import { mount } from '@vue/test-utils';
import { createTestingPinia } from '@pinia/testing';
import ElementPlus from 'element-plus';

vi.mock('@/composables/journal/use-game-list', () => ({
  default: vi.fn().mockReturnValue({ loading: true, games: ref([]) }),
}));

vi.mock('vue-router', () => ({
  useRoute: vi.fn(),
  useRouter: vi.fn(() => ({
    push: vi.fn(),
  })),
}));

import JournalView from '@/features/journal/JournalView.vue';

describe('JournalView', () => {
  test('should mount component', async () => {
    expect(JournalView).toBeTruthy();

    const wrapper = mount(JournalView, {
      global: {
        plugins: [
          ElementPlus,
          createTestingPinia({
            initialState: {
              search: {
                results: [],
                loading: false,
              },
            },
          }),
        ],
        stubs: {
          RouterView: true,
          HSearchInput: true,
        },
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
