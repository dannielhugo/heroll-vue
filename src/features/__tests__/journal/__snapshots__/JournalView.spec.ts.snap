// Vitest Snapshot v1

exports[`JournalView > should mount component 1`] = `
"<div class=\\"journal-view__search\\" data-v-d8aeafea=\\"\\">
  <h-search-input-stub placeholder=\\"Find a game\\" data-v-d8aeafea=\\"\\"></h-search-input-stub>
</div>
<div class=\\"journal-container el-loading-parent--relative\\" data-v-06c35f8d=\\"\\" data-v-d8aeafea=\\"\\" loading-number=\\"1\\">
  <div class=\\"game-list\\" data-v-ef199a12=\\"\\" data-v-06c35f8d=\\"\\"></div>
  <transition-stub>
    <div class=\\"el-loading-mask\\" style=\\"display: none;\\">
      <div class=\\"el-loading-spinner\\"><svg class=\\"circular\\" viewBox=\\"25 25 50 50\\">
          <circle class=\\"path\\" cx=\\"50\\" cy=\\"50\\" r=\\"20\\" fill=\\"none\\"></circle>
        </svg>
        <!---->
      </div>
    </div>
  </transition-stub>
</div>
<div role=\\"pagination\\" aria-label=\\"pagination\\" class=\\"el-pagination journal-view__pagination\\" data-v-d8aeafea=\\"\\"><button type=\\"button\\" class=\\"btn-prev is-first\\" disabled=\\"\\" aria-disabled=\\"true\\"><i class=\\"el-icon\\"><svg viewBox=\\"0 0 1024 1024\\" xmlns=\\"http://www.w3.org/2000/svg\\">
        <path fill=\\"currentColor\\" d=\\"M609.408 149.376 277.76 489.6a32 32 0 0 0 0 44.672l331.648 340.352a29.12 29.12 0 0 0 41.728 0 30.592 30.592 0 0 0 0-42.752L339.264 511.936l311.872-319.872a30.592 30.592 0 0 0 0-42.688 29.12 29.12 0 0 0-41.728 0z\\"></path>
      </svg></i></button>
  <ul class=\\"el-pager\\">
    <li class=\\"is-active is-disabled number\\" aria-current=\\"true\\" tabindex=\\"-1\\"> 1 </li>
    <!--v-if-->
    <!--v-if-->
    <!--v-if-->
  </ul><button type=\\"button\\" class=\\"btn-next\\" disabled=\\"\\" aria-disabled=\\"true\\"><i class=\\"el-icon\\"><svg viewBox=\\"0 0 1024 1024\\" xmlns=\\"http://www.w3.org/2000/svg\\">
        <path fill=\\"currentColor\\" d=\\"M340.864 149.312a30.592 30.592 0 0 0 0 42.752L652.736 512 340.864 831.872a30.592 30.592 0 0 0 0 42.752 29.12 29.12 0 0 0 41.728 0L714.24 534.336a32 32 0 0 0 0-44.672L382.592 149.376a29.12 29.12 0 0 0-41.728 0z\\"></path>
      </svg></i></button><span class=\\"el-pagination__jump is-last\\" disabled=\\"true\\">Go to <div class=\\"el-input el-input--small is-disabled el-pagination__editor is-in-pagination\\"><!-- input --><!-- prepend slot --><!--v-if--><div class=\\"el-input__wrapper\\"><!-- prefix slot --><!--v-if--><input class=\\"el-input__inner\\" min=\\"1\\" max=\\"1\\" type=\\"number\\" disabled=\\"\\" autocomplete=\\"off\\" tabindex=\\"0\\"><!-- suffix slot --><!--v-if--></div><!-- append slot --><!--v-if--></div> </span>
</div>
<router-view-stub data-v-d8aeafea=\\"\\"></router-view-stub>"
`;
